alternator/executor.cc

```diff
@@ -2660,18 +2665,53 @@ void executor::describe_single_item(const cql3::selection::selection& selection,
     }
 }
 
+/*
+ * \brief calculate the length in bytes of a row.
+ *
+ * To be compatible with DynamoDB API we need to calculate RCU.
+ * RCU calculation is based on the data read include the the column
+ * names and data.
+ */
+static uint64_t get_result_row_length(const cql3::selection::selection& selection,
+        const std::vector<managed_bytes_opt>& result_row) {
+    uint64_t result = 0;
+    const auto& columns = selection.get_columns();
+    auto column_it = columns.begin();
+    for (const managed_bytes_opt& cell : result_row) {
+        if (cell) {
+            if ((*column_it)->name_as_text() != executor::ATTRS_COLUMN_NAME) {
+                result += (*column_it)->name_as_text().length();
+                result += cell->size();
+            } else {
+                auto deserialized = attrs_type()->deserialize(*cell);
+                auto keys_and_values = value_cast<map_type_impl::native_type>(deserialized);
+                for (const auto& entry : keys_and_values) {
+                    result += value_cast<sstring>(entry.first).length();
+                    // We add one extra byte compared to DynamoDB for the bytes lengh
+                    result += value_cast<bytes>(entry.second).length() - 1;
+                }
+            }
+        }
+        ++column_it;
+    }
+    return result;
+}
```

Comment:
It's too wasteful to do this in a separate loop - it will cause a measurable slowdown. This should be measured in an existing loop.

Also, it needs to be done on the replica, where it can measure tombstones.

Reply:
We are measuring what the user would have gotten, given no filters, not what we read and how.
We are after the value a user would have gotten if it was a DynamoDB.

Reply:
@avikivity can you please look at @amnonh 's answers and provide an explicit "approve" or "request changes" judgement? Thanks.